projects/plugins/boost/app/modules/cache/pre-wordpress/Boost_Cache_Utils.php

```diff
@@ -5,15 +5,20 @@
 
 namespace Automattic\Jetpack_Boost\Modules\Page_Cache\Pre_WordPress;
 
+define( 'JBCACHE_ALL', 1 ); // delete all files and directories in a given directory, recursively.
+define( 'JBCACHE_FILE', 2 ); // delete a single file or recursively delete a single directory in a given directory.
+define( 'JBCACHE_FILES', 3 ); // delete all files in a given directory.
+
 class Boost_Cache_Utils {
 	/**
 	 * Recursively delete a directory.
 	 * @param string $dir - The directory to delete.
-	 * @param bool   $recurse - If false, only delete the files in the directory, do not recurse into subdirectories.
+	 * @param bool   $filter - The filter to use. JBCACHE_FILES to delete all files in the given directory. JBCACHE_ALL to delete everything in the given directory, recursively. JBCACHE_FILE to delete a single file or directory in the given directory.
+	 * @param string $filename - The filename to delete. Only used when $filter is JBCACHE_FILE.
 	 * @return bool|WP_Error
 	 */
-	public static function delete_directory( $dir, $recurse = true ) {
-		error_log( "delete directory: $dir " . (int) $recurse ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
+	public static function delete_directory( $dir, $filter, $filename = '' ) {
```

Comment:
Fyi, I have created a new `Filesystem_Utils` since we are doing many filesystem operations. This probably belongs there. It might be a good idea to move the method to the other class in a separate PR if this PR is already too changes.

Reply:
Since this can actually also delete a single file, perhaps we can just call it `delete`?

Reply:
Good ideas. Thanks.

Reply:
I renamed the function, but I'll leave moving it to another PR.